internal/appsec/emitter/httpsec/roundtripper.go

```diff
@@ -0,0 +1,56 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016 Datadog, Inc.
+
+package httpsec
+
+import (
+	"context"
+	"net/http"
+	"sync"
+
+	"gopkg.in/DataDog/dd-trace-go.v1/appsec/events"
+	"gopkg.in/DataDog/dd-trace-go.v1/internal/appsec/dyngo"
+	"gopkg.in/DataDog/dd-trace-go.v1/internal/appsec/emitter/httpsec/types"
+	"gopkg.in/DataDog/dd-trace-go.v1/internal/appsec/listener"
+	"gopkg.in/DataDog/dd-trace-go.v1/internal/log"
+)
+
+var badInputContextOnce sync.Once
+
+func RoundTrip(ctx context.Context, request *http.Request, rt http.RoundTripper) (*http.Response, error) {
+	url := request.URL.String()
+	opArgs := types.RoundTripOperationArgs{
+		URL: url,
+	}
+
+	parent, _ := ctx.Value(listener.ContextKey{}).(dyngo.Operation)
```

Comment:
```suggestion
	parent = listener.FromContext(ctx)
```

even though `listener` doesn't sound like the right place for it, it would already by better to make it public and use it from here too.